source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -174,6 +179,13 @@ class ComponentInterface : public NodeT {
       bool fixed_topic = false, const std::string& default_topic = ""
   );
 
+  /**
+   * @brief Function to daemonize the callback function given in input.
```

Comment:
Just a suggestion to improve this docstring
```suggestion
   * @brief Add a periodic callback function.
   * @details The provided function is evaluated periodically at the component step period. 
```
And, as discussed, perhaps `add_periodic_callback` is better than `add_daemon` as a function name.